aws/lambda/log-classifier/src/bedrock.rs

```diff
@@ -62,25 +90,42 @@ pub fn create_log_snippet<'a>(log: Log, error_line: &str, num_lines: usize) -> V
         }
     }
 
-    // If the error line is not found, return nothing
+    // If the error line is not found, return an empty vector
     if !found_error_line {
         return Vec::new();
     }
 
-    // if snippet is too larget shrink it to the size of num_lines by cutting off the beginning
+    // If the snippet is too large, shrink it to the size of num_lines by cutting off the beginning
     if snippet.len() > num_lines {
-        snippet = snippet.split_at(snippet.len() - num_lines).1.to_vec();
+        snippet = snippet.split_at(snippet.len() - num_lines - 1).1.to_vec();
     }
 
     snippet
 }
-pub async fn make_query(input_text: &String) -> Option<String> {
+
+/// Makes a query to an AI model using the provided log snippet.
+///
+/// This function creates a log snippet, sends it to two different AI models,
+/// and validates the output. If a valid response is found, it is returned.
+///
+/// # Arguments
+///
+/// * `log` - A reference to the Log structure containing the full log
+/// * `error_line` - A string slice that should be contained in the error line
+/// * `num_lines` - The maximum number of lines to include in the log snippet
+///
+/// # Returns
+///
+/// An Option<String> containing the validated AI response, or None if no valid response was found.
+pub async fn make_query(log: &Log, error_line: &str, num_lines: usize) -> Option<String> {
     let model_id_primary = "anthropic.claude-3-haiku-20240307-v1:0";
     let model_id_secondary = "anthropic.claude-3-5-sonnet-20240620-v1:0";
 
-    let response = make_bedrock_call(input_text, model_id_primary).await;
+    let log_snippet = create_log_snippet(log, error_line, num_lines);
+    let input_text = log_snippet.join("\n");
 
-    // validate the response
+    // Try the primary model
+    let response = make_bedrock_call(&input_text, model_id_primary).await;
```

Comment:
Nit: It looks nicer to put this part into a function with the model name as  one of its inputs